plugins/wasmembed/secure-enclave.cpp

```
@@ -0,0 +1,624 @@
+            {
+                int thisSize;
+                switch (elemType)
+                {
+                case type_varstring:
+                    thisSize = strlen((const char *)inData) + 1;
+                    break;
+                case type_string:
+                    thisSize = *(size32_t *)inData + sizeof(size32_t);
+                    break;
+                case type_unicode:
+                    thisSize = (*(size32_t *)inData) * sizeof(UChar) + sizeof(size32_t);
+                    break;
+                case type_utf8:
+                    thisSize = rtlUtf8Size(*(size32_t *)inData, inData + sizeof(size32_t)) + sizeof(size32_t);
+                    break;
+                default:
+                    rtlFail(0, "wasmembed: Unsupported parameter type");
+                    break;
+                }
+                inData += thisSize;
+                numElems++;
+            }
+            inData = (const byte *)setData;
+        }
+        else
+            numElems = totalBytes / elemSize;
+
+        std::vector<wasmtime::Val> memIdxVar;
+        int32_t memIdx;
+
+        switch (typecode)
+        {
+        case type_boolean:
+            memIdxVar = wasmStore->callRealloc(wasmName, {0, 0, 1, (int32_t)numElems});
+            memIdx = memIdxVar[0].i32();
+            break;
+        default:
+            rtlFail(0, "wasmembed: Unsupported parameter type");
+            break;
+        }
+
+        auto mem = wasmStore->getData(wasmName);
+        size32_t thisSize = elemSize;
+        for (int idx = 0; idx < numElems; idx++)
+        {
+            switch (typecode)
+            {
+            case type_boolean:
+                mem[memIdx + idx] = *(bool *)inData;
+                break;
+            default:
+                rtlFail(0, "v8embed: Unsupported parameter type");
+                break;
+            }
+            inData += thisSize;
+        }
+        args.push_back(memIdx);
+        args.push_back(numElems);
```

Strictly should throw unsupported.